### ISSUE TITLE
Am I populating this Picker in the right way?

### ISSUE THREAD

**Issue by ajtruckle** [xamarin/Xamarin.Forms#13747]
XAML:

```
<Picker x:Name="StylesPicker" Title="Select stylesheet" HorizontalOptions="FillAndExpand" />
<Button x:Name="InstallStyleButton" Text="Select" Clicked="InstallStyleButton_Clicked" Padding="10,10,10,10"  />
```

In the **code behind** I have a **public property**:

`public ObservableCollection<string> StylesList { get; set; }`

In the **constructor** I populate this property and **attach** it to the **Picker**:

```
string[] stylesArray = Directory.GetFiles("/Users/Shared/VisitsRota.MacOS/Styles", "ElderlyInfirm-Schedule-*.xsl")
                                      .Select(file => Path.GetFileName(file)).ToArray<string>();
StylesList = new ObservableCollection<string>(stylesArray);
StylesPicker.ItemsSource = StylesList;
```
The way I understand this MVVM concept, the Window itself should be responsible for filling this Picker with a list of available files on the Mac and not a ViewModel. The ViewModel will kick in for the persistence of the selected picker value (not yet coded).

The, I added the button handler. I understand that using async void is not a good idea and I have to research this and find a better way:

```
        async private void InstallStyleButton_Clicked(object sender, EventArgs e)
        {
            var customFileType =
                new FilePickerFileType(new Dictionary<DevicePlatform, IEnumerable<string>>
                {
                    {DevicePlatform.macOS, new[] {"xsl"} }
                });

            var pickResult = await FilePicker.PickAsync(new PickOptions
            {
                FileTypes = customFileType,
                PickerTitle = "Select template to install"

            });

            if(pickResult != null)
            {
                StylesList.Add(pickResult.FileName);
            }
        }
```

This works as I expect and it adds the selected filet the list. The code is incomplete as I now need to verify that the file name is not already in the list, and I have to then copy the in to the new location. I have to investigate that.

In short, it was correct in this instance to populate age picker via code behind property and not view model. Correct?

**Comment by ajtruckle** [comment]
This I worked this out now - hopefully!

https://stackoverflow.com/a/66179628/2287576
